qiita_db/support_files/patches/40.sql

```diff
@@ -0,0 +1,118 @@
+-- Mar 12, 2016
+-- Add software_type table. This new table allows us to have different types of
+-- plugins. Here, we will introduce a new type, the "type plugin", and define
+-- the previous type as the "processing plugin" type. The new group "type plugin"
+-- define plugins that do not perform any processing on the artifacts but they
+-- are able to validate that they're correct and generate they summary page.
+-- These new plugins are special. They are not directly visible by the end Qiita
+-- user by they are useful to plugin developers so they do not need to re-define
+-- types if they already exist. This way, multiple plugins can share the same
+-- type of artifacts without depending in another "processing" plugin.
+
+-- Create the new table to hold the software types
+CREATE TABLE qiita.software_type (
```

Comment:
Correct - the software table holds the actual plugins (in its inception was not like that hence the awful name) - I've added an issue (https://github.com/biocore/qiita/issues/1691) so we change the names.

There are 2 types of plugins, the ones who define new artifact types, and others that apply transformations to those artifact. They don't operate directly upon another plugin, but they require it to be installed in the system in order for such type to be defined in the system (and hence the artifact definition plugin be added to the system).